### PR TITLE
Push the rest of the development of the map

### PR DESCRIPTION
**Pull request by Polo150202**
Add the bottom scaffold to display the infos of the location using the Places API inspired by @mtv47's code. Add some design update on the map and the components of the Map.

### CODE REVIEW

File: app/src/main/java/com/github/se/wanderpals/ui/screens/trip/Map.kt [hunk 1]
```diff
@@ -117,24 +157,28 @@ fun Map(mapViewModel: MapViewModel, client: PlacesClient) {
...
   Box(Modifier.fillMaxSize().testTag("mapScreen")) {
     // search bar for searching a location on the map and pass it to geocoder
     if (enabled) {
       DockedSearchBar(
           // align the search bar to the top left corner with padding
-          modifier = Modifier.testTag("searchBar").align(Alignment.TopCenter).padding(top = 8.dp),
+          modifier =
+              Modifier.align(Alignment.TopCenter)
+                  .padding(top = 8.dp)
+                  .shadow(15.dp, shape = RoundedCornerShape(40.dp))
```
Comment (mtv47) [inline app/src/main/java/com/github/se/wanderpals/ui/screens/trip/Map.kt:168]: Nice graphical improve and is greatly appreciated

File: app/src/main/java/com/github/se/wanderpals/ui/screens/trip/Map.kt [hunk 2]
```diff
@@ -145,35 +189,48 @@ fun Map(mapViewModel: MapViewModel, client: PlacesClient) {
...
-            // suggestions
+            // if the search bar is expanded when clicked modify the search text, show the
+            // list of suggestions
             if (expanded) {
               listOfPropositions.forEach {
                 val primaryText = it.getPrimaryText(null).toString()
                 val placeId = it.placeId
-                val placeFields = listOf(Place.Field.LAT_LNG)
+
+                val placeFields =
```
Comment (mtv47) [inline app/src/main/java/com/github/se/wanderpals/ui/screens/trip/Map.kt:222]: As discussed during the research that we have done together in person, the tags are interesting and relevant.

File: app/src/main/java/com/github/se/wanderpals/ui/screens/trip/Map.kt [hunk 3]
```diff
@@ -191,42 +264,122 @@ fun Map(mapViewModel: MapViewModel, client: PlacesClient) {
...
+
     GoogleMap(
-        modifier = Modifier.matchParentSize().testTag("googleMap"),
-        properties = MapProperties(mapType = MapType.NORMAL),
-        uiSettings = uiSettings,
+        modifier = Modifier.testTag("googleMap").matchParentSize(),
+        properties =
+            MapProperties(
+                mapType = MapType.NORMAL,
+                mapStyleOptions = MapStyleOptions.loadRawResourceStyle(context, R.raw.map_style)),
```
Comment (mtv47) [inline app/src/main/java/com/github/se/wanderpals/ui/screens/trip/Map.kt:273]: Nice style change, makes it less generic

File: app/src/main/java/com/github/se/wanderpals/ui/screens/trip/Map.kt [hunk 4]
```diff
@@ -191,42 +264,122 @@ fun Map(mapViewModel: MapViewModel, client: PlacesClient) {
...
+              }
+            }
+
+            if (placeOpeningHours.isNotEmpty()) {
+              Spacer(modifier = Modifier.height(16.dp))
+              val listOfDays = placeOpeningHours.removePrefix("[").removeSuffix("]").split(", ")
+              listOfDays.forEach {
+                Text(
+                    modifier = Modifier.align(Alignment.CenterHorizontally).padding(bottom = 8.dp),
+                    text = it)
```
Comment (mtv47) [inline app/src/main/java/com/github/se/wanderpals/ui/screens/trip/Map.kt:375]: Check that "it" isn't null, I don't know why but "null" can be displayed:
image

**Review by mtv47 — CHANGES_REQUESTED** [review]
Overall great job and very nice graphical improvement

File: app/src/main/java/com/github/se/wanderpals/ui/screens/trip/Map.kt (hunk 4 above)
Comment (Polo150202) [reply app/src/main/java/com/github/se/wanderpals/ui/screens/trip/Map.kt:375]: Okay, I will fix it

**Review by mtv47 — APPROVED** [review]
Looks Good!

**Pull request merged**